bot/exts/moderation/incidents.py

```diff
@@ -402,10 +515,106 @@ async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent) ->
 
     @Cog.listener()
     async def on_message(self, message: discord.Message) -> None:
-        """Pass `message` to `add_signals` if and only if it satisfies `is_incident`."""
+        """
+        Pass `message` to `add_signals` and `extract_message_links` if it satisfies `is_incident`.
+
+        If the message (`message`) is an incident report, then run it through `extract_message_links`
+        to get all the message link embeds (embeds which contain information about that particular
+        link).These message link embeds are then sent into the channel.
+
+        Also passes the message into `add_signals` if the message is an incident.
+        """
         if is_incident(message):
             await add_signals(message)
 
+            webhook_embed_list = await extract_message_links(message, self.bot)
+            if webhook_embed_list:
+                await self.send_message_link_embeds(webhook_embed_list, message, self.incidents_webhook)
+
+    @Cog.listener()
+    async def on_message_edit(self, msg_before: discord.Message, msg_after: discord.Message) -> None:
+        """
+        Pass `msg_after` to `extract_message_links` and edit `msg_before` webhook msg.
+
+        Deletes cache value (`message_link_embeds_cache`) of `msg_before` if it exists and removes the
+        webhook message for that particular link from the channel.
+
+        If the message edit (`msg_after`) is a incident then run it through `extract_message_links`
+        to get all the message link embeds (embeds which contain information about that particular
+        link), this message link embeds are then sent into the channel.
+
+        The edited message is also passed into `add_signals` if it is a incident message.
+        """
+        if is_incident(msg_after):
+            webhook_embed_list = await extract_message_links(msg_after, self.bot)
+            webhook_msg_id = await self.message_link_embeds_cache.get(msg_before.id)
+
+            if webhook_msg_id:
+                await self.incidents_webhook.edit_message(
+                    message_id=webhook_msg_id,
+                    embeds=[x for x in webhook_embed_list if x is not None],
+                )
```

Comment:
Please use a better name than `x`. Why not `embed`?

According to the return type annotation of `extract_message_links`, the rvalue can be None. This code seems to assume that the rvalue is always iterable.

Reply:
Resolved in efd100b and e2bf492.